app/com/linkedin/drelephant/tuning/engine/SparkHBTParamRecommender.java

```diff
@@ -360,4 +430,62 @@ private int getMemoryIncreaseForGC() {
       return 0;
     }
   }
+
+  /**
+   * This method suggest three parameters : spark.dynamicAllocation.minExecutors, spark.dynamicAllocation.maxExecutors
+   *  and spark.executor.instances
+   * For now we are changing these parameters only in case we are changing number of executor core
+   */
+  private void suggestNumExecutorParameters() {
+    if (suggestedCore != lastRunExecutorCore) {
+      if (lastRunDynamicAllocationMinExecutors != null) {
+        suggestedDynamicAllocationMinExecutors =
+            (int) Math.ceil(lastRunDynamicAllocationMinExecutors * lastRunExecutorCore / suggestedCore);
+      }
+      if (lastRunDynamicAllocationMaxExecutors != null) {
+        suggestedDynamicAllocationMaxExecutors =
+            (int) Math.ceil(lastRunDynamicAllocationMaxExecutors * lastRunExecutorCore / suggestedCore);
+      }
+      if (lastRunExecutorInstances != null) {
```

Comment:
Should we set spark.dynamicAllocation.initialExecutors as well? Basically, initial executors in Spark are max(min executors, initial executors, executor instances). Probably set all params the same instead calcuating for each one based on earlier config.
Refer to Spark code in method getDynamicAllocationInitialExecutors in Utils.scala https://github.com/apache/spark/blob/7bf0794651f4d11547325539ebf7131a57ee1ba2/core/src/main/scala/org/apache/spark/util/Utils.scala

Reply:
Now we are only setting executor instances.

Reply:
And in LinkedIn specific case, executor instances is used for initial executor as well as max executors.